clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java

```diff
@@ -486,10 +488,14 @@ public class ConsumerConfig extends AbstractConfig {
                                         CommonClientConfigs.METRIC_REPORTER_CLASSES_DOC)
                                 .define(KEY_DESERIALIZER_CLASS_CONFIG,
                                         Type.CLASS,
+                                        ConfigDef.NO_DEFAULT_VALUE,
+                                        new ConfigDef.NonNullValidator(),
```

Comment:
Isn't this redundant? Why do we need a `NonNullValidator` when we've explicitly marked this property as having no default value?

Reply:
As described in JIRA, expect to throw `ConfigException` instead of `NullPointerException` when user passes in `null`

Reply:
Yes, I understand the motivation; I was wondering if we would automatically catch the case of null values when we'd defined the property with no default value.

To answer my own question, an explicitly-set `null` in the original `Map<String, Object>` is not the same as having no value set at all, and since users can programmatically instantiate `KafkaConsumer` instances, it's possible that they may have specified `null` for the `key.deserializer` property and others like it. As a result, it is necessary to guard against null values for this property, either with preflight validation or downstream manual checks.

Reply:
thanks